ftw/table/catalog_source.py

```diff
@@ -46,6 +48,18 @@ def datetime_compare(x, y):
                   reverse=reverse)
 
 
+def remove_rid_from_brains(rid, brains):
+    if rid not in brains._seq:
+        return
+
+    # ._seq is actually a LazyValues instance with a ._seq of 3-tuples, the middle one of which is the rid
+    position = sum(1 for lazyvalue in takewhile(lambda lazyvalue: rid not in lazyvalue, brains._seq._seq))
```

Comment:
Hmm, I pasted that into a `Plone=4.3.4` with a `Products.ZCatalog=2.13.27`:
I don't have a `brains._seq._seq` there it does not contain a tuple, so we have a completely different data structure.

`ftw.table` needs to be compatible with both versions.

We could make a more robust implementation, maybe checking for a `brains._seq._seq` and switching implementation, or we could try to re-instantiate the `LazyMap` object with the filtered rids - altough this may also be tricky with diverging implementations.

What do you think?

Reply:
We're lucky the divergence only affects the internal implementation of how to remove a brain based on the rid from the `LazyMap`.

Adding a third test run where the currently standing implementation of this PR fails would be a good start.

Reply:
I'm at least not spotting anything obvious on the change log.

https://github.com/zopefoundation/Products.ZCatalog/blob/2.13/CHANGES.txt

So the change is a Ploneside thing?

Reply:
Oh, we have `Products.ZCatalog=3.0.4` in `opengever.core`.

https://github.com/4teamwork/opengever.core/commit/9878734e91f51c01bca6aee7e861f23cfb8714e9
https://github.com/4teamwork/kgs/commit/a16b5e791da0100d292eb6415a0b7f2d8ae6bbb1

And seems we only pinned this starting in `2017.0` - anyone remember what for?